server/odc-service/src/main/java/com/oceanbase/odc/service/db/session/DBSessionService.java

```diff
@@ -74,25 +76,31 @@ public List<OdcDBSession> list(@NonNull ConnectionSession session) {
         return jdbcOperations.query("SHOW FULL PROCESSLIST", new OdcDBSessionRowMapper());
     }
 
-    public List<SessionIdKillSql> getKillSql(@NonNull ConnectionSession session, @NonNull List<String> sessionIds,
+    public List<SessionIdKillSql> getKillSql(@NonNull ConnectionSession session,
+            @NonNull List<SessionIdSvpIp> sessionIds,
             String closeType) {
         List<OdcDBSession> allSession = list(session);
-        Map<String, String> sessionId2SvrpIp =
-                allSession.stream().collect(
-                        Collectors.toMap(OdcDBSession::getSessionId,
-                                s -> MoreObjects.firstNonNull(s.getSvrIp(), "")));
+        Map<String, List<OdcDBSession>> sidMap = allSession.stream().collect(
```

Comment:
If we can assume `sessionId` is unique, how about returning `Map<String, OdcDBSession>` which is more understandable, instead of `Map<String, List<OdcDBSession>>`. Then when you get `OdcDBSession` from the map, you do not need to `get(0)` from the session list which is kind of tricky.

Reply:
from ob4.2.5 and 4.3.x, sessionId here can't guaranteed to be unique.

Reply:
The logic is kind of tricky here, we may add some comments... Like why sessionId is not unique and explain why get(0) here makes sense.